torch/distributed/checkpoint/state_dict.py

```diff
@@ -858,7 +858,7 @@ def _load_optim_state_dict(
                 optim_state_dict = FSDP.optim_state_dict_to_load(
                     model, optim, optim_state_dict
                 )
-        elif info.broadcast_from_rank0:
+        elif (info.broadcast_from_rank0 or info.full_state_dict):
```

Comment:
This is not correct. If `info.broadcast_from_rank0` is False, then all ranks have the full_state_dict. We should extract the `full_state_dict` logic from here and make it as a standalone API. And then `info.broadcast_from_rank0` will be come `broadcasting` + `full_state_dict`.

You also need to do a similar thing for model state_dict.

Reply:
Thank you for the comment, now I switch the condition to full_state_dict
And the broadcasting action are conducted under the condition of (broadcast_from_rank0 and rank=0)